Source/WebCore/css/parser/CSSPropertyParserConsumer+Appearance.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CSSPropertyParserConsumer+Appearance.h"
+
+namespace WebCore {
+namespace CSSPropertyParserHelpers {
+
+static bool consumeFunctionBody(CSSParserTokenRange& range)
+{
+    while (!range.atEnd() && range.peek().type() != RightParenthesisToken) {
+        const auto& token = range.consumeIncludingWhitespace();
+        if (token.type() == FunctionToken && !consumeFunctionBody(range))
+            return false;
+    }
+    return true;
+}
+
+static bool consumeFunctionPatameter(CSSParserTokenRange& range, CSSPropertyID property, CSS::PropertyParserState& state, CSS::PropertyParserResult& result)
+{
+    auto parameterRange = range;
+
+    while (!range.atEnd() && range.peek().type() != CommaToken) {
+        const auto& token = range.consumeIncludingWhitespace();
+        if (token.type() == FunctionToken && !consumeFunctionBody(range))
+            break;
+    }
```

Comment:
I'm not sure what is the best way to write the parsing code for this with all the new CSS parsing infra but @weinig surely has ideas. 

I would do something more like the code in `isValidDashedFunction()` which is similar in that it parses a comma separated argument list of arbitrary CSS values. The `validateArgument` lambda would be replaced with the code to actually parse the argument ranges.